controlplane/src/core/repositories/UserRepository.ts

```diff
@@ -85,4 +85,24 @@ export class UserRepository {
       await orgRepo.addOrganizationMemberRoles({ memberID: insertedMember.id, roles: ['member'] });
     });
   }
+
+  // checks if the user has the right roles to perform the operation
+  public async checkUserAccess(input: { userID?: string; organizationID: string; rolesToBe: MemberRole[] }) {
+    const orgRepo = new OrganizationRepository(this.db);
+
+    if (!input.userID) {
+      return true;
+    }
+
+    const roles = await orgRepo.getOrganizationMemberRoles({
+      userID: input.userID,
+      organizationID: input.organizationID,
+    });
+    for (const role of input.rolesToBe) {
+      if (roles.includes(role)) {
```

Comment:
Wouldn't that behave in correct fail if someone requires `['member', 'admin']`? So it will skip admin in that case.

Reply:
they dont require both the roles

Reply:
it just looks for one of them

Reply:
So it an OR check?

Reply:
yep